### PR TITLE
Update moab version

### PR DESCRIPTION
**Pull request by AI-Pranto**
- Update moab version
- Install CMake by pip rather than apt because It installs **CMake 3.10** version in ubuntu 18.04.

### CODE REVIEW

File: docker/ubuntu_18.04-dev.dockerfile
Comment (gonuke) [inline docker/ubuntu_18.04-dev.dockerfile]: We are intentionally using the version of `nuc_data_make` that is in the path and has been installed, rather than the version that is in the repository.  This is meant to double check that the process is carried out consistently with the version that has been installed.

File: docker/ubuntu_18.04-dev.dockerfile
Comment (gonuke) [inline docker/ubuntu_18.04-dev.dockerfile]: we are intentionally only pip-installing python packages and instead relying on the system to install CMake

**Review by gonuke — CHANGES_REQUESTED** [review]
Hello @AI-Pranto - Thanks for this contribution.  It is a good idea for us to update to MOAB5.2 for our testing.

This dockerfile is only intended for use in our continuous testing, and not for users to build PyNE on their own system.

If you are looking for dockerfiles for users to use for their own system - you can look at our install_scripts repository (https://github.com/pyne/install_scripts).  We recently updated these to MOAB 5.2 and take pull requests for updates there as well.

**Comment by gonuke** [comment]
@bam241 can you look at updating our CI images with MOAB5.2 once this is merged?

File: docker/ubuntu_18.04-dev.dockerfile
Comment (AI-Pranto) [reply docker/ubuntu_18.04-dev.dockerfile]: My mistake @gonuke. I didn't consider it `ENV PATH $HOME/.local/bin:$PATH`. Should I revert the change @gonuke?

**Comment by gonuke** [comment]
I think we can merge this if we revert the change on how we install cmake

**Comment by kkiesling** [comment]
@AI-Pranto - looks like this just needs a conflict resolved in the CHANGELOG and then it should be good to go

**Comment by gonuke** [comment]
I went ahead and used the web editor to resolve this minor conflict.

**Review by gonuke — APPROVED** [review]

**Comment by bam241** [comment]
Testing this right now.

**Comment by bam241** [comment]
@gonuke as you approved it, I'll merge the PR when I confirm that the docker image are working locally.

I'll then push the new image on dockerhub

**Comment by AI-Pranto** [comment]
Sorry for the late response @kkiesling, I was busy with other things. Thanks @gonuke.

**Comment by bam241** [comment]
I think` MOAB 5.2.0` is breaking our workflow.
at least it breaks our tests:
```
nosetests test_activation_responses.py
/usr/bin/h5diff
/root/.local/lib/python3.6/site-packages/pyne/alara.py:1063: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  (response_name, np.float64, data_length)
./usr/bin/h5diff
..Loading file geom.h5m
Initializing the GeomQueryTool...
Using faceting tolerance: 0.001
Building acceleration data structures...
The file blank_mesh.h5m has been saved to disk.
Do not delete this file; it is needed by activation_responses.py step2.

Decay_heat step1 complete, run ALARA with the command:
>> alara alara_inp > output.txt
/usr/bin/h5diff
/root/.local/lib/python3.6/site-packages/pyne/alara.py:1063: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  (response_name, np.float64, data_length)
Writing decay_heat for decay time: 1E3 s
Activation_response step2 complete.
attribute: <default of </tstt/tags/GLOBAL_ID>> and <default of </tstt/tags/GLOBAL_ID>>
1 differences found
F
======================================================================
FAIL: test_activation_responses.test_activation_responses_script
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/pyne/tests/test_activation_responses.py", line 270, in test_activation_responses_script
    _activation_responses_test_step2(activation_responses_run_dir)
  File "/opt/pyne/tests/test_activation_responses.py", line 257, in _activation_responses_test_step2
    assert_equal(diff_flag5, 0)
AssertionError: 256 != 0

----------------------------------------------------------------------
Ran 4 tests in 1.889s

FAILED (failures=1)```

**Comment by bam241** [comment]
So we confirmed that we are not yet compatible with MOAB 5.2.0.

The PR looks good as is for me, but we definitely need to fix our compatibility with it first (#1325)  before we are able to merge this.

File: CHANGELOG.rst
Comment (gonuke) [inline CHANGELOG.rst:264]: Let's leap frog to 5.2.1 since it's now available.
```suggestion
   * Update moab version 5.1.0 >> 5.2.1 (#1312)
```

File: docker/ubuntu_18.04-dev.dockerfile
Comment (gonuke) [inline docker/ubuntu_18.04-dev.dockerfile]: version 5.2.1 is not available as a branch, just a tag
```suggestion
        && git clone --single-branch -b 5.2.1 https://bitbucket.org/fathomteam/moab \
```

File: docker/ubuntu_18.04-dev.dockerfile
Comment (gonuke) [inline docker/ubuntu_18.04-dev.dockerfile]: Is this necessary?

**Review by gonuke — CHANGES_REQUESTED** [review]
Thanks @AI-Pranto - it looks like we've resolved the compatibility issue with MOAB 5.2.x, so we can probably go ahead with this.

Please rebase against the `develop` branch and move your Changelog entry into the "Next Version"  section near the top.

File: docker/ubuntu_18.04-dev.dockerfile
Comment (AI-Pranto) [reply docker/ubuntu_18.04-dev.dockerfile]: @gonuke, It's not necessary. I will revert this change.

**Comment by AI-Pranto** [comment]
@gonuke I rebase against `develop` branch but look like something is wrong. I never did rebase before. I will close this PR and go for another PR.

**Comment by AI-Pranto** [comment]
Now I understand where I did wrong during the rebase against  `develop` branch. 
> Please rebase against the `develop` branch

Because after rebase, all the changes will go into the `pyne:0.7.0-rc` branch not in the `develop` branch. Isn't it better to go for a new PR @gonuke?

**Comment by gonuke** [comment]
Sorry for the rebasing confusion...

For a new PR, it is best to use the `develop` branch - we have abandoned the `0.7.0-rc` branch as that release is complete.  Under normal circumstances, all PR's should go to `develop`.

**Pull request closed without merging**
